js/data/bucket.js

```diff
@@ -58,8 +58,6 @@ class Bucket {
                 options.featureIndex.insert(feature, this.index);
```

Comment:
This code isn't used for symbols. `SymbolBucket` has [it's own definition of `populate`](https://github.com/mapbox/mapbox-gl-js/blob/b9d95095c63a8e0193b6ee391ab308e75e1ecea3/js/data/bucket/symbol_bucket.js#L138).